examples/Building-and-deploying-multi-stage-RecSys/01-Building-Recommender-Systems-with-Merlin.ipynb

```diff
@@ -7,7 +7,7 @@
    "metadata": {},
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #5.    item_embs_df["item_id"] = item_features["item_id"]
</pre><p>I am skeptical that this step is correct.  This is not doing correct mapping.  if we are trying to replace encoded items with their original raw item_ids, we should use <code>unique.item_id.parquet</code>. so we can do something like </p><p> </p><pre class="ql-syntax" spellcheck="false">raw_item_ids = pd.read_parquet('./categories/unique.item_id.parquet')
item_embs_df["item_id"] = raw_item_ids["item_id"]  
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/NVIDIA-Merlin/Merlin/pull/1022/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='70'/>

Reply:
<p>In addition, are we sure we want to save item_embeddings for raw item_ids? Arent we supposed to do KNN search after retrieval stage based on the encoded items? If yes, we should then get rid of the steps above.</p>

Reply:
You need to save the embeddings with raw_item_ids because those embeddings should correspond to raw data. This way when you do the lookup in the second notebook you are getting the actual items. Those items are then processed through the item subworkflow to get the features for the model to rank the items. Suppose you wanted to use that similarity search with a different feature engineering workflow, if you had placed the data with the item ids already categorified that would not work. For the item_id to raw_item_ids you are correct that it is not correctly merged. However, I pose the question why are we sorting IDs by default? I see that we use it in the code for categorify to handle making na/nulls the first items in the dataframe... but as a user I think I expect to see the categories given in a first come first serve order. Unless I elect to use freq_threshold or kwarg that changes the strategy. Anyway I will fix for now.